### ISSUE TITLE
Allow arbitrary use of keys by MetaMask

### ISSUE THREAD

**Issue by DB-DBDBD** [MetaMask/metamask-extension#6498]
Multiple sites now let me use MetaMask to sign a message, like one that proves I'm the owner of my address, but I cannot do this directly. If I could mark a body of text (say this text) and let MetaMask sign it for me (just generate a signature text text, nothing on-chain), I can paste it to my email/website as proof that the owner of this address signed this text. This will be a big step towards making the address/keys the cornerstone of identity 3.0. MetaMask of course will also need to be able to take the signature and verify it against the original text, for the other party. 

This can be expanded, when public keys are available, for me to be able to encrypt a message for the owner of a specific address (again, if he published his public key and email), and use regular email, for example, as a safe way to send (encrypted) information. All from within the browser. 

Basically just letting users use keys available to them in any way they wish (signing, encrypting, validating or decrypting), directly on texts or files.

Thanks for considering
Danny

**Comment by bdresser** [comment]
relates to #1190

**Comment by 1blockologist** [comment]
relates to #5018 that I reported/requested August 2018

**Comment by logvik** [comment]
Relates to https://github.com/MetaMask/eth-simple-keyring/pull/15
We need to have this implementation in the package because it is used for all keys in Metamask. After confirmation for this pull request, I will be able to publish implementation for encrypting/decrypting messages in Metamask.
